watson-developer-cloud/discovery-components
# bug: Clear all button working with collection facet

#### What do these changes do/fix?
Previously, the root `Clear all` button for the SearchFacets component was not appearing when collectionFacets were selected, and it was also not clearing collectionFacets when it was clicked. These changes make it so that this `Clear all` button interacts with the collectionFacets component as it does the other facets components.

related to issues: 
https://github.ibm.com/watson-discovery/disco-widgets/issues/427
https://github.ibm.com/watson-discovery/disco-widgets/issues/428

#### How do you test/verify these changes?
I added some tests for this interaction.
You can also test in the example app with a project that has collections. Select a collection and see that the `Clear all` button appears. Click the `Clear all` button and see that it deselects the collections that you selected.

#### Have you documented your changes (if necessary)?
I added comments to props that were added, and I also left a comment in the SearchFacets component regarding what we should change when Carbon MultiSelect offers more control over selection. I also opened a Carbon enhancement request here: https://github.com/carbon-design-system/carbon/issues/4812.

#### Are there any breaking changes included in this pull request?
I don't believe so since new props are only being added to the CollectionFacets component, but not to the top-level SearchFacets component.
